### PR TITLE
feat: internal link

### LINKED ISSUE #264: Tooltips - Infobulles - les rendre responsive
🇫🇷 
Actuellement, le contenu de l'infobulle s'affiche sur une ligne unique, donc si le contenu est trop riche, une partie de l'infobulle est tronquée

### PR DESCRIPTION
**Pull request by ddecrulle**
### Summary

This pull request enhances the Lunatic library by introducing a new feature: the ability to extract internal links from Markdown text. Additionally, improvements have been made to the display of tooltips, restoring it to the v1 Lunatic behavior.

### Changes Made

1. **Internal Link Extraction:**
   - Added functionality to extract internal links from Markdown text.
   - Internal links are now detected based on the classic Markdown link pattern starting with "/". Clients can customize the `RouterLink` component using the `custom` props to provide their own link.

2. **Tooltip Display:**
   - Corrected the display of tooltips to align with the behavior observed in Lunatic v1.
   - To create a tooltip, use the Markdown syntax: `[word](url 'tooltip content')`. If no URL is provided, use `[word](. 'tooltip content')`.

### Usage

To get the internal link extraction feature and the corrected tooltip display:

1. Update Lunatic to this branch.
2. In the Orchestrateur, customize the `RouterLink` component using the `custom` props and provide your own link.
3. Verify that internal links within Markdown text, starting with "/", are properly extracted and displayed.
4. Ensure tooltips now display as expected.

### Related Issues

Closes: Inseefr/Stromae#264

### Screenshots

Capture d’écran 2023-11-21 à 14 35 08

Capture d’écran 2023-11-21 à 14 35 13

### Test

The tests use a snapshot and the Link component `useId` hook, which is not mocked, making them fragile.

### CODE REVIEW

File: src/components/commons/components/md-label/router-link.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { ComponentProps } from 'react';
+import createCustomizableLunaticField from '../../create-customizable-field';
+
+type Props = Omit<ComponentProps<'a'>, 'href'> & { to: string };
+
+const RouterLink = (props: Props) => {
+	const { to, children, onClick, ref, ...rest } = props;
+	return (
+		<a href={to} onClick={onClick} ref={ref} {...rest}>
```
Comment (Grafikart) [inline src/components/commons/components/md-label/router-link.tsx]: La destructuration ne sert à rien pour onClick ref et children

File: tests/setup.ts [hunk 2]
```diff
@@ -14,3 +14,9 @@ afterEach(() => {
 (global as any).structuredClone = (val: any) => {
 	return JSON.parse(JSON.stringify(val));
 };
+
+global.ResizeObserver = vi.fn().mockImplementation(() => ({
```
Comment (Grafikart) [inline tests/setup.ts:19]: Ajouter un commentaire pour expliquer ce code.

File: src/components/commons/components/md-label/link.tsx [hunk 3]
```diff
@@ -1,55 +1,52 @@
...
+
+	const id = useId();
+
+	const LinkComponent: React.FC<{
+		to?: string;
+		href?: string;
+		id: string;
+		'data-tooltip-id'?: string;
+	}> = (() => {
+		if (href.trim().startsWith('/')) {
```
Comment (Grafikart) [inline src/components/commons/components/md-label/link.tsx:23]: On ne pourrait pas avoir des liens relatif aussi genre "./blog"

File: src/components/commons/components/md-label/link.tsx [hunk 4]
```diff
@@ -1,55 +1,52 @@
...
-				data-tip={title}
-				data-multiline
-				className="field-md"
+		<>
+			<LinkComponent
+				{...linkProps}
+				{...(title
+					? { 'data-tooltip-id': `tooltip-${id}`, className: 'link-md' }
+					: {})}
 			>
```
Comment (Grafikart) [inline src/components/commons/components/md-label/link.tsx]: autant calculé toutes les props en amont car niveau lisibilité c'est pas top là :(

**Review by Grafikart — CHANGES_REQUESTED** [review]

File: src/components/commons/components/md-label/link.tsx (hunk 3 above)
Comment (ddecrulle) [reply src/components/commons/components/md-label/link.tsx:23]: Je vois pas trop de cas d’usage pour ça t’en as en tête ?

File: src/components/commons/components/md-label/router-link.tsx (hunk 1 above)
Comment (ddecrulle) [reply src/components/commons/components/md-label/router-link.tsx]: Pour children je pense que c'est bien de le laisser.

**Review by laurentC35 — APPROVED** [review]

**Pull request merged**